### ISSUE TITLE
grad_norm nan

### ISSUE THREAD

**Issue by Echo0125** [modelscope/ms-swift#2280]
单机8卡训练qwen2-vl 7b正常step不会报错，更换2机16卡后，首先是worker机报warning
```bash
It looks like you are trying to rescale already rescaled images. If the input images have pixel values between 0 and 1, set `do_rescale=False` to avoid rescaling them again.
```
随后，loss突然增大，grad_norm变成nan：
```bash
{'loss': 1.80130756, 'acc': 0.56179339, 'grad_norm': 37.75608444, 'learning_rate': 0.0, 'memory(GiB)': 62.94, 'train_speed(iter/s)': 0.007755, 'epoch': 0.0, 'global_step/max_steps': '1/2169', 'percentage': '0.05%', 'elapsed_time': '1m 22s', 'remaining_time': '2d 1h 43m 27s'}
{'loss': 663.56591797, 'acc': 0.14326748, 'grad_norm': nan, 'learning_rate': 3.84e-06, 'memory(GiB)': 73.64, 'train_speed(iter/s)': 0.010765, 'epoch': 0.0, 'global_step/max_steps': '5/2169', 'percentage': '0.23%', 'elapsed_time': '6m 58s', 'remaining_time': '2d 2h 15m 22s'}
```

机器为H800, config为
```bash
NUM_NODE=$1
RANK=$2
MASTER_ADDR_=$3

echo "MASTER_ADDR=$MASTER_ADDR_"

CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
NNODES=${NUM_NODE} \
NODE_RANK=${RANK} \
MASTER_ADDR=${MASTER_ADDR_} \
NPROC_PER_NODE=8 \
swift sft \
  --model_type qwen2-vl-7b-instruct \
  --model_id_or_path ./checkpoints/Qwen2-VL-7B-Instruct \
  --sft_type full \
  --dataset ./train.json \
  --ddp_backend nccl \
  --warmup_ratio 0.03 \
  --weight_decay 0.1 \
  --deepspeed default-zero2 \
  --batch_size 1 \
  --gradient_accumulation_steps 8 \
  --save_strategy epoch \
  --use_flash_attn True \
  --eval_steps 500 \
  --val_dataset ./val.json \
  --save_total_limit 1 \
  --num_train_epochs 1 \
  --learning_rate 2e-5 \
  --lazy_tokenize true \
  --save_only_model true \
  --tuner_backend swift \
  --dtype bf16 \
  --save_steps 5000
  ```

**Comment by yufei1900** [comment]
遇到了类似现象，loss还是下降的，gradnorm很高，最后nan。internvl2-8b也遇到了。
![screenshot-20241025-170356](https://github.com/user-attachments/assets/0077ff08-3c63-4141-8bce-311ff757509d)

**Comment by conheaven** [comment]
![image](https://github.com/user-attachments/assets/825f4846-a880-4af3-b630-1aa05d1a2824)
我这情况怎么办

**Comment by wqf321** [comment]
请问解决了吗

**Comment by conheaven** [comment]
解决了，之前是用多卡会出现这种错误，只用一个显卡的话就不会



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2025年01月22日 23:20 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [modelscope/ms-swift] grad_norm nan (Issue #2280) |

请问解决了吗

**Comment by yangqiancheng-yuan** [comment]
多卡目前有解决方案吗？求大佬指点

**Comment by JacksonShengWei** [comment]
> 多卡目前有解决方案吗？求大佬指点

sry,之后一直都是单卡，还没试过多卡，ms-swift最近不是有大更新，还是不行吗？

**Comment by yangqiancheng-yuan** [comment]
> > 多卡目前有解决方案吗？求大佬指点
> 
> sry,之后一直都是单卡，还没试过多卡，ms-swift最近不是有大更新，还是不行吗？

昨晚的大更新不行，一小时前我看代码又更新了，好像是针对性解决，我看看行不行

**Comment by wqf321** [comment]
> > > 多卡目前有解决方案吗？求大佬指点
> > 
> > 
> > sry,之后一直都是单卡，还没试过多卡，ms-swift最近不是有大更新，还是不行吗？
> 
> 昨晚的大更新不行，一小时前我看代码又更新了，好像是针对性解决，我看看行不行

请问有结论了吗

**Comment by yangqiancheng-yuan** [comment]
> > > > 多卡目前有解决方案吗？求大佬指点
> > > 
> > > 
> > > sry,之后一直都是单卡，还没试过多卡，ms-swift最近不是有大更新，还是不行吗？
> > 
> > 
> > 昨晚的大更新不行，一小时前我看代码又更新了，好像是针对性解决，我看看行不行
> 
> 请问有结论了吗

新更新的代码还是有问题，可能需要超参搜索一波，怀疑是kl的超参设置问题，你可以群里冒一下泡，我们一起搜索一下参数

**Comment by Jintao-Huang** [comment]
感觉是 机器的问题

我这里是正常的

**Comment by Jintao-Huang** [comment]
> > > 多卡目前有解决方案吗？求大佬指点
> > 
> > 
> > sry,之后一直都是单卡，还没试过多卡，ms-swift最近不是有大更新，还是不行吗？
> 
> 昨晚的大更新不行，一小时前我看代码又更新了，好像是针对性解决，我看看行不行

那个fix是GRPO的

**Comment by yangqiancheng-yuan** [comment]
> 感觉是 机器的问题
> 
> 我这里是正常的

能给个参数行示例吗？我1:1实验一下，确认一下是不是机器问题，多谢。如果可以的话，辛苦给一下一些关键库函数的版本。

**Comment by yangqiancheng-yuan** [comment]
> > > > 多卡目前有解决方案吗？求大佬指点
> > > 
> > > 
> > > sry,之后一直都是单卡，还没试过多卡，ms-swift最近不是有大更新，还是不行吗？
> > 
> > 
> > 昨晚的大更新不行，一小时前我看代码又更新了，好像是针对性解决，我看看行不行
> 
> 那个fix是GRPO的

抱歉，看错issue了，以为这是grpo那个。

**Comment by wqf321** [comment]
> 感觉是 机器的问题
> 
> 我这里是正常的

我改成了zero3又正常了，之前用的是zero2就会出现gradnorm突然增长的现象，我看主楼用的也是zero2。

**Comment by wqf321** [comment]
> 感觉是 机器的问题
> 
> 我这里是正常的

2机8卡梯度与loss正常的运行命令`FPS=1 \
MAX_PIXELS=101920 \
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
NNODES=2 \
NODE_RANK=$NODE_RANK \
MASTER_ADDR=$MASTER_ADDR \
NPROC_PER_NODE=8 \
MASTER_PORT=$MASTER_PORT swift sft \
  --model_type qwen2_5_vl \
  --model $model_path \
  --train_type full \
  --freeze_vit true \
  --dataset $train_dataset_path \
  --logging_steps 1 \
  --deepspeed zero3 \
  --dataloader_num_workers 4 \
  --num_train_epochs 2 \
  --save_steps 200 \
  --eval_steps 50 \
  --learning_rate 1e-05 \
  --save_total_limit 5 \
  --per_device_train_batch_size 2 \
  --gradient_accumulation_steps 2 \
  --torch_dtype bfloat16 \
  --max_length 8192 \
  --attn_impl flash_attn \
  --gradient_checkpointing true \

sleep 100000000`

**Comment by wqf321** [comment]
> 感觉是 机器的问题
> 
> 我这里是正常的

使用zero2的2x8卡启动命令：
`FPS=1 \
MAX_PIXELS=101920 \
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
NNODES=2 \
NODE_RANK=$NODE_RANK \
MASTER_ADDR=$MASTER_ADDR \
NPROC_PER_NODE=8 \
MASTER_PORT=$MASTER_PORT swift sft \
  --model_type qwen2_5_vl \
  --model $model_path \
  --train_type full \
  --freeze_vit true \
  --dataset $train_dataset_path \
  --logging_steps 1 \
  --deepspeed zero2 \
  --dataloader_num_workers 4 \
  --num_train_epochs 2 \
  --save_steps 200 \
  --eval_steps 50 \
  --learning_rate 1e-05 \
  --save_total_limit 5 \
  --per_device_train_batch_size 2 \
  --gradient_accumulation_steps 2 \
  --torch_dtype bfloat16 \
  --max_length 8192 \
  --attn_impl flash_attn \
  --gradient_checkpointing true \
`

**Comment by wqf321** [comment]
> 感觉是 机器的问题
> 
> 我这里是正常的

使用zero2的2x8卡启动命令：`FPS=1 \
MAX_PIXELS=101920 \
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
NNODES=2 \
NODE_RANK=$NODE_RANK \
MASTER_ADDR=$MASTER_ADDR \
NPROC_PER_NODE=8 \
MASTER_PORT=$MASTER_PORT swift sft \
  --model_type qwen2_5_vl \
  --model $model_path \
  --train_type full \
  --freeze_vit true \
  --dataset $train_dataset_path \
  --logging_steps 1 \
  --deepspeed zero2 \
  --dataloader_num_workers 4 \
  --num_train_epochs 2 \
  --save_steps 200 \
  --eval_steps 50 \
  --learning_rate 1e-05 \
  --save_total_limit 5 \
  --per_device_train_batch_size 2 \
  --gradient_accumulation_steps 2 \
  --torch_dtype bfloat16 \
  --max_length 8192 \
  --attn_impl flash_attn \
  --gradient_checkpointing true \`

结果如下：`{'loss': 2.0916357, 'token_acc': 0.47775947, 'grad_norm': 6.08488035, 'learning_rate': 1e-05, 'memory(GiB)': 65.89, 'train_speed(iter/s)': 0.014105, 'epoch': 0.0, 'global_step/max_steps': '1/2046', 'percentage': '0.05%', 'elapsed_time': '14s', 'remaining_time': '8h 17m 1s'}
Train:   0%|                                                                                                                                                     | 1/2046 [00:14<8:16:53, 14.58s/it][h264 @ 0x7fc648030400] mmco: unref short failure
{'loss': 2.0832541, 'token_acc': 0.52479339, 'grad_norm': nan, 'learning_rate': 1e-05, 'memory(GiB)': 65.89, 'train_speed(iter/s)': 0.025912, 'epoch': 0.0, 'global_step/max_steps': '2/2046', 'percentage': '0.10%', 'elapsed_time': '20s', 'remaining_time': '5h 55m 32s'}
{'loss': 0.0, 'token_acc': 0.0, 'grad_norm': nan, 'learning_rate': 1e-05, 'memory(GiB)': 68.85, 'train_speed(iter/s)': 0.036217, 'epoch': 0.0, 'global_step/max_steps': '3/2046', 'percentage': '0.15%', 'elapsed_time': '26s', 'remaining_time': '5h 1m 1s'}
{'loss': 0.0, 'token_acc': 0.0, 'grad_norm': nan, 'learning_rate': 1e-05, 'memory(GiB)': 68.85, 'train_speed(iter/s)': 0.0453, 'epoch': 0.0, 'global_step/max_steps': '4/2046', 'percentage': '0.20%', 'elapsed_time': '31s', 'remaining_time': '4h 32m 10s'}
{'loss': 0.0, 'token_acc': 0.0, 'grad_norm': nan, 'learning_rate': 1e-05, 'memory(GiB)': 68.85, 'train_speed(iter/s)': 0.053166, 'epoch': 0.0, 'global_step/max_steps': '5/2046', 'percentage': '0.24%', 'elapsed_time': '37s', 'remaining_time': '4h 16m 42s'}
{'loss': 0.0, 'token_acc': 0.0, 'grad_norm': nan, 'learning_rate': 1e-05, 'memory(GiB)': 68.85, 'train_speed(iter/s)': 0.060258, 'epoch': 0.01, 'global_step/max_steps': '6/2046', 'percentage': '0.29%', 'elapsed_time': '43s', 'remaining_time': '4h 5m 8s'}
{'loss': 0.0, 'token_acc': 0.0, 'grad_norm': nan, 'learning_rate': 1e-05, 'memory(GiB)': 68.85, 'train_speed(iter/s)': 0.066873, 'epoch': 0.01, 'global_step/max_steps': '7/2046', 'percentage': '0.34%', 'elapsed_time': '48s', 'remaining_time': '3h 54m 47s'}`

改成zero3就正常了

![Image](https://github.com/user-attachments/assets/bfe9a3cb-cce8-4ab5-b05e-eedb26e133d4)

**Comment by zengxingchen** [comment]
遇到了类似的问题，grad_norm变为nan，loss和 token_acc变为0。
通用通过从zero2改为zero3解决了。
